roachtest: changes to reuse spot VM clusters

nameisbhaskar opened the pull request:
We have a provision to use spot VMs for running tests by setting a flag. This flag is set when the test is selected for running, after a cluster is selected. This ends up creating a new cluster with spot VM every time a test is marked to use spot VM.
The change is to set the testspec to use spot Vm before the tests gets registered to the worker.

Fixes: #116778
Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/122875)
<!-- Reviewable:end -->

nameisbhaskar committed:
roachtest: changes to reuse spot VM clusters

We have a provision to use spot VMs for running tests by setting a flag. This flag is set when the test is selected for running, after a cluster is selected. This ends up creating a new cluster with spot VM every time a test is marked to use spot VM.
The change is to set the testspec to use spot Vm before the tests gets registered to the worker.

Fixes: #116778
Epic: none
